src/DependencyManagement/DependencyManager.cs

```diff
@@ -161,19 +179,49 @@ internal void Initialize(FunctionLoadRequest request)
 
                     // Validate the module version.
                     string majorVersion = GetMajorVersion(version);
-                    string latestVersion = DependencyManagementUtils.GetModuleLatestSupportedVersion(name, majorVersion);
-                    ValidateModuleMajorVersion(name, majorVersion, latestVersion);
 
-                    // Before installing the module, check the path to see if it is already installed.
-                    var moduleVersionFolderPath = Path.Combine(DependenciesPath, name, latestVersion);
-                    if (!Directory.Exists(moduleVersionFolderPath))
+                    // Try to connect to the PSGallery via C# API to get the latest module supported version.
```

Comment:
Please remove "via C# API" - this is incorrect, but also the API details do not seem to be relevant at this place.

Reply:
Comment updated.